File: src/scripting_way/witch.lua
witch = {
    name="Lucia",
    interact=function(witch, player)
        if(player.hp == player.maxHp) then
            witch:say("Hi! Come back when you're wounded.")
        else
            player.hp = player.maxHp
            witch:say("Hi! Let me heal you.")
        end
    end
}

File: src/oop_way/Witch.cpp
#include "Witch.h"
#include <iostream>

void Witch::interact(Player* player) {
    if (player->getHp() == player->getMaxHp()) {
        std::cout << name << ": Hi! Come back when you're wounded." << std::endl;
    } else {
        std::cout << name << ": Hi! Let me heal you." << std::endl;
        player->setHp(player->getMaxHp());
    }
}